homeassistant/components/update/__init__.py

```diff
@@ -216,6 +216,13 @@ def installed_version(self) -> str | None:
         """Version installed and in use."""
         return self._attr_installed_version
 
+    def _default_to_device_class_name(self) -> bool:
+        """Return True if an unnamed entity should be named by its device class.
+
+        For updates this is True if the entity has a device class.
+        """
+        return self.device_class is not None
```

Comment:
Do we need to check if the device class is an actual `UpdateDeviceClass`?

Reply:
I added in the description the counterparts for button, number and sensor entities. None of them checks if the device class has the correct type.

Reply:
> Do we need to check if the device class is an actual UpdateDeviceClass?

The device class may not be anything else. If it is something else, it would be a bug.

Reply:
But why do we check this in SensorEntity?

Reply:
Because the sensor entity has the history of having free-form device classes.

Reply:
Ah understood, thanks!